dotnet/src/Connectors/Connectors.Memory.Redis/RedisMemoryStore.cs

```diff
@@ -28,13 +28,58 @@ public sealed class RedisMemoryStore : IMemoryStore
     /// <summary>
     /// Create a new instance of semantic memory using Redis.
     /// </summary>
-    /// <param name="database">The database of the redis server.</param>
-    /// <param name="vectorSize">Embedding vector size</param>
-    public RedisMemoryStore(IDatabase database, int vectorSize)
+    /// <param name="database">The database of the Redis server.</param>
+    /// <param name="vectorSize">Embedding vector size, defaults to 1536</param>
+    /// <param name="vectorIndexAlgorithm">Indexing algorithm for vectors, defaults to "HNSW"</param>
+    /// <param name="vectorType">Vector type, defaults to "FLOAT32"</param>
+    /// <param name="vectorDistanceMetric">Metric for measuring vector distances, defaults to "COSINE"</param>
+    /// <param name="queryDialect">Query dialect, must be 2 or greater for vector similarity searching, defaults to 2</param>
+    public RedisMemoryStore(IDatabase database, int vectorSize = 1536,
+        VectorIndexAlgorithms vectorIndexAlgorithm = VectorIndexAlgorithm,
+        VectorTypes vectorType = VectorType,
+        VectorDistanceMetrics vectorDistanceMetric = VectorDistanceMetric,
+        int queryDialect = QueryDialect)
     {
         this._database = database;
         this._vectorSize = vectorSize;
         this._ft = database.FT();
+        // Default to HNSW if parsing error occurs
+        if (!Enum.TryParse<Schema.VectorField.VectorAlgo>(vectorIndexAlgorithm.ToString(), out this._vectorIndexAlgorithm))
```

Comment:
Do we want to go to a default know value when we can't parse, or do we want to throw an exception so that the user knows right away that something is wrong?

Reply:
Good point, an exception would be more beneficial. Another missing exception handling is the vector size.